documentation/examples/dynamic-sync-table/typeform.ts

```diff
@@ -0,0 +1,259 @@
+    for (let field of form.fields) {
+      // Format the field name into a valid property name.
+      let name = getPropertyName(field.title);
+      // Generate a schema for the field and add it to the set of properties.
+      properties[name] = getSchema(field);
+      // Mark the property as featured (included in the table by default).
+      featured.push(name);
+    }
+
+    // Assemble the schema for each row.
+    let schema = coda.makeObjectSchema({
+      properties: properties,
+      primary: primary,
+      id: id,
+      featured: featured,
+      identity: {
+        name: "FormResponse",
+        dynamicUrl: formUrl,
+      },
+    });
+
+    // Return an array schema as the result.
+    return coda.makeSchema({
+      type: coda.ValueType.Array,
+      items: schema,
+    });
+  },
+
+  // The formula that syncs the records.
+  formula: {
+    name: "SyncResponses",
+    description: "Sync the form responses",
+    parameters: [],
+    execute: async function ([], context) {
+      let formUrl = context.sync.dynamicUrl;
+
+      // Retrieve the token to continue from, if any.
+      let pageToken = context.sync.continuation?.token || null;
+
+      // Construct the API URL.
+      let url = coda.withQueryParams(formUrl + "/responses", {
+        page_size: PageSize,
+        before: pageToken,
+      });
+
+      // Fetch a page of responses.
+      let response = await context.fetcher.fetch({
+        method: "GET",
+        url: url,
+        // Disable HTTP caching, so we always get the latest results.
+        cacheTtlSecs: 0,
+      });
+      let formResponses = response.body.items;
+
+      // Reformat each response to match the schema.
+      let rows = [];
+      for (let formResponse of formResponses) {
+        // Include the metadata common to all forms.
+        let row = {
+          submitted: formResponse.submitted_at,
+          responseId: formResponse.response_id,
+        };
+
+        // For each answer, add it to the row.
+        for (let answer of formResponse.answers) {
+          let value = getValue(answer);
+          // Store the answer based on the field ID. See the use of `fromKey` in
+          // `getSchema` below.
+          row[answer.field.id] = value;
+        }
+        rows.push(row);
+      }
+
+      // Determine if we need to fetch more pages.
+      let continuation;
+      if (formResponses.length > 0) {
+        // Continue onward from the token of the last response.
+        let lastToken = formResponses[formResponses.length - 1].token;
+        continuation = {
+          token: lastToken,
+        };
+      }
+
+      // Return the rows and the continuation, if any.
+      return {
+        result: rows,
+        continuation: continuation,
+      }
+    },
+  },
+});
+
+// Get metadata about a form given it's URL.
+async function getForm(context, url) {
+  let response = await context.fetcher.fetch({
+    method: "GET",
+    url: url,
+    cacheTtlSecs: 0,
```

Comment:
Might want to add a note about disabling the cache.

Reply:
Done